libs/mintycore/src/main/java/com/solanamobile/mintyfresh/mintycore/usecase/PerformMintUseCase.kt

```diff
@@ -201,7 +201,12 @@ class PerformMintUseCase @Inject constructor(
                         return@withContext
                     }
 
-                _mintState.value = MintState.Complete(transactionSignature)
+                if (isConfirmed) {
+                    _mintState.value = MintState.Complete(transactionSignature)
+                } else {
+                    _mintState.value =
+                        MintState.Error(context.getString(R.string.transaction_confirmation_failure_message))
```

Comment:
Are we sure that not being confirmed is an error state? Or does the previous call wait for the proper confirmation.

Also, does it make the minting process longer as it waits for conf?

Reply:
Made a few more changes which does this 

1. Actually throw an error and fail confirmation. 
2. Exponential delay when checking for confirmations. .5s then 1s, 2s, 4s .. so on until timeout.
3. Initial delay between sending transaction and checking for confirmations.
4. Use delay(long) instead of while loop.
5. Fix the condition in while loop. (this was always false before so we never went inside the loop to actually wait for the confirmations.) thats why it never worked before.